src/mc_rbdyn/RobotConverter.cpp

```diff
@@ -0,0 +1,175 @@
+    for(const auto & joint : inputRobot.mb().joints())
+    {
+      const auto & jname = joint.name();
+      if(outputRobot.hasJoint(jname)
+         && outputRobot.mb().joint(static_cast<int>(outputRobot.jointIndexByName(jname))).dof() == joint.dof())
+      {
+        commonJointIndices_.emplace_back(inputRobot.jointIndexByName(jname), outputRobot.jointIndexByName(jname));
+      }
+    }
+  }
+
+  if(config_.copyEncoderValues_ || config_.copyEncoderVelocities_ || config_.encodersToOutMbc_)
+  { // Common actuated joints and its index in output mbc
+    commonEncoderToJointIndices_.reserve(inputRobot.refJointOrder().size());
+    commonEncoderIndices_.reserve(inputRobot.refJointOrder().size());
+    const auto & rjo = inputRobot.refJointOrder();
+    for(size_t i = 0; i < rjo.size(); ++i)
+    {
+      const auto & jname = rjo[i];
+      if(outputRobot.hasJoint(jname))
+      {
+        const auto & oRjo = outputRobot.refJointOrder();
+        auto it = std::find(oRjo.begin(), oRjo.end(), jname);
+        if(it != oRjo.end())
+        {
+          commonEncoderIndices_.emplace_back(i, std::distance(oRjo.begin(), it));
+        }
+
+        auto mbcIndex = outputRobot.jointIndexByName(jname);
+        commonEncoderToJointIndices_.emplace_back(i, mbcIndex);
+      }
+    }
+  }
+
+  if(config_.copyEncoderValues_)
+  {
+    outEncoderValues_.resize(outputRobot.refJointOrder().size());
+  }
+
+  if(config_.copyEncoderVelocities_)
+  {
+    outEncoderVelocities_.resize(outputRobot.refJointOrder().size());
+  }
+
+  if(config_.computeMimics_)
+  {
+    for(const auto & m : outputRobot.mb().joints())
+    {
+      if(m.isMimic())
+      {
+        auto mainIndex = outputRobot.jointIndexByName(m.mimicName());
+        auto mimicIndex = outputRobot.jointIndexByName(m.name());
+        mimicJoints_.emplace_back(mainIndex, mimicIndex);
+      }
+    }
+  }
+}
+
+void RobotConverter::convert(const mc_rbdyn::Robot & inputRobot, mc_rbdyn::Robot & outputRobot)
+{
+  if(first_)
+  {
+    precompute(inputRobot, outputRobot);
+    first_ = false;
+  }
+
+  // Copy the encoders into outputRobot
+  const auto & encoders = inputRobot.encoderValues();
+  if(encoders.size() == inputRobot.refJointOrder().size())
+  {
+    if(config_.encodersToOutMbc_)
+    {
+      for(const auto & indices : commonEncoderToJointIndices_)
+      {
+        outputRobot.mbc().q[indices.second][0] = encoders[indices.first];
+      }
+    }
+    if(config_.copyEncoderValues_)
+    {
+      for(const auto & indices : commonEncoderIndices_)
+      {
+        outEncoderValues_[indices.second] = encoders[indices.first];
+      }
+      outputRobot.encoderValues(outEncoderValues_);
+    }
+    if(config_.copyEncoderVelocities_)
+    {
+      for(const auto & indices : commonEncoderIndices_)
+      {
+        outEncoderVelocities_[indices.second] = encoders[indices.first];
+      }
+      outputRobot.encoderVelocities(outEncoderVelocities_);
+    }
```

Comment:
```suggestion
    if(config_.copyEncoderValues_)
    {
      outputRobot.encoderValues(encoders);
    }
    if(config_.copyEncoderVelocities_)
    {
      outputRobot.encoderVelocities(inputRobot.encoderVelocities());
    }
```
note: in addition to the change, `outEncoderVelocities_` was setting the values from the encoders and not the encoders' velocities

Reply:
You are not guaranteed that the `outputRobot` has the same `refJointOrder` as the main robot, so you cannot just copy the vector ;)